sdk/storage/azure_storage_blob/src/models/blob_properties.rs

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+use azure_core::{
+    headers::{
+        FromHeaders, HeaderName, Headers, BLOB_ACCESS_TIER, BLOB_TYPE, CREATION_TIME, LEASE_STATE,
+        LEASE_STATUS, SERVER_ENCRYPTED,
+    },
+    Etag, LeaseStatus,
+};
+
+use crate::models::{AccessTier, BlobType, LeaseState};
+
+pub const CONTENT_LENGTH: HeaderName = HeaderName::from_static("content-length");
+pub const CONTENT_MD5: HeaderName = HeaderName::from_static("content-md5");
+pub const CONTENT_TYPE: HeaderName = HeaderName::from_static("content-type");
+pub const ETAG: HeaderName = HeaderName::from_static("etag");
+pub const LAST_MODIFIED: HeaderName = HeaderName::from_static("last-modified");
+pub const BLOB_ACCESS_TIER_INFERRED: HeaderName =
+    HeaderName::from_static("x-ms-access-tier-inferred");
+
+/// Properties of an Azure Storage blob.
+///
+#[derive(Clone, Default, Debug)]
+pub struct BlobProperties {
+    pub access_tier_inferred: Option<bool>,
+    pub access_tier: Option<AccessTier>,
+    pub blob_type: Option<BlobType>,
+    pub content_length: Option<i64>,
+    pub content_md5: Option<String>,
+    pub content_type: Option<String>,
+    pub creation_time: Option<String>,
+    pub etag: Option<Etag>,
+    pub last_modified: Option<String>,
+    pub lease_state: Option<LeaseState>,
+    pub lease_status: Option<LeaseStatus>,
+    pub server_encrypted: Option<bool>,
+}
+
+impl FromHeaders for BlobProperties {
+    type Error = url::ParseError;
+
+    fn header_names() -> &'static [&'static str] {
```

Comment:
I was a little confused when trying to pattern match this-- is this supposed to be an exhaustive list of the possible optional headers? For now, I only had the plain-text version of the headers I had to create a new `const` for.

Reply:
This is from our older, unofficial code. Frankly, I'm not sure what purpose it serves. Do you see any usage throughout the rest of the repo (Shift+F12 by default in VSCode)? Maybe we could get rid of it if there's no use case currently.

Reply:
Gotcha. I mostly cribbed this from the test example here:
https://github.com/Azure/azure-sdk-for-rust/blob/048717f4f5fc758e85769663d3279ff67437ebde/sdk/typespec/typespec_client_core/src/http/headers/mod.rs#L347-L361

But after some more digging and reading the documentation, I see this is supposed to representative of the possible headers that the `from_headers` function expects to parse (so should be **exhaustive**). I will apply this fix.